core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -213,6 +213,11 @@ oppia.factory('oppiaPlayerService', [
                 _explorationId));
           }
           loadedExploration.then(function(data) {
+            // Update exploration's states field.
+            for (var stateName in data.exploration.states) {
```

Comment:
@seanlip That makes a lot of sense. I think we can do that. I'll see how we can work on it.